src/utils/object_pool.hpp

```diff
@@ -0,0 +1,245 @@
+//========================================================================================
+// (C) (or copyright) 2022. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#ifndef UTILS_OBJECT_POOL_HPP_
+#define UTILS_OBJECT_POOL_HPP_
+
+#include <math.h>
+
+#include <cstdint>
+#include <iostream>
+#include <memory>
+#include <stack>
+#include <type_traits>
+#include <unordered_map>
+#include <utility>
+
+#include <Kokkos_Core.hpp>
```

Comment:
With all the Kokkos annotation in this file, was this inspired by some Kokkos code?

Reply:
No, but that was all required to get it to inherit from a `Kokkos::View`sensibly. Might have been better to just go with composition.